src/use_debounced_callback/index.jsx

```diff
@@ -1,11 +1,18 @@
 import debounce from 'lodash/debounce';
-import { useMemo } from 'react';
-
-const useDebouncedCallback = (func, delay) => (
-  useMemo(
-    () => debounce(func, delay),
-    [func, delay],
-  )
-);
+import { useEffect, useState } from 'react';
+
+const useDebouncedCallback = (func, delay) => {
+  const [instance, setInstance] = useState(undefined);
+
+  useEffect(() => {
```

Comment:
We cannot use `useMemo` here because react might decide to recalculate memorized values as it likes - would cause weird issues in our case.
https://reactjs.org/docs/hooks-reference.html#usememo

Reply:
nice

Reply:
Please explain how it should help

Reply:
yes you're right it doesn't fit. do we need the memo at all? and with state? what about not memoing/canceling last calls on render?

Reply:
Yeah we need the memo. We need to control when to start a new queue. I tried using ref instead of state but the test suite already suffers from the missing render cycle on deps change.

```js
import { useEffect, useRef, useState } from 'react';

const EMPTY_SYMBOL = Symbol('empty');

const useStableMemo = (fn, teardownFn, deps) => {
  const firstRun = useRef(true);
  const memorizedRef = useRef(EMPTY_SYMBOL);
  if (memorizedRef.current === EMPTY_SYMBOL) {
    memorizedRef.current = fn();
  }

  useEffect(() => {
    let localMemorized;

    if (firstRun.current) {
      firstRun.current = false;
      localMemorized = memorizedRef.current;
    } else {
      localMemorized = fn();
      memorizedRef.current = localMemorized;
    }

    return teardownFn.bind(undefined, localMemorized);
  }, deps);

  return memorizedRef.current;
};

export default useStableMemo;

```